### PR TITLE
{App Config} Update raised errors in app config command module

### PR DESCRIPTION
**Pull request by albertofori**
**Related command**
`az appoconfig`

**Description**
This change is being added to updated the raised errors to more specific errors in `azure.cli.core.azclierror` instead of the base `CLIError` class. Follow up to [PR #24400](https://github.com/Azure/azure-cli/pull/24400)

**Testing Guide**

**History Notes**

[App Config] `az appconfig`: Update raised errors in app config command module

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
App Config

**Comment by albertofori** [comment]
@zhoxing-ms @avanigupta Could you please take a look at this follow up PR for updating the raised errors in the app configuration command module. This change covers all other files in the module. 
Thanks!

File: src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py [hunk 1]
```diff
@@ -335,7 +339,7 @@ def __read_kv_from_config_store(azconfig_client,
                                                                              accept_datetime=datetime,
                                                                              fields=query_fields)
     except HttpResponseError as exception:
-        raise CLIError('Failed to read key-value(s) that match the specified key and label. ' + str(exception))
+        raise HTTPError('Failed to read key-value(s) that match the specified key and label. ' + str(exception))
```
Comment (avanigupta) [inline src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py:342]: Should this be `AzureResponseError`?

Comment (albertofori) [reply src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py:342]: You are right, I think `HTTPResponseError`s should raise `AzureResponseError`

File: src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py [hunk 2]
```diff
@@ -367,7 +369,7 @@ def set_key(cmd,
         except ResourceNotFoundError:
             logger.debug("Key '%s' with label '%s' not found. A new key-value will be created.", key, label)
         except HttpResponseError as exception:
-            raise CLIError("Failed to retrieve key-values from config store. " + str(exception))
+            raise CLIErrors.HTTPError("Failed to retrieve key-values from config store. " + str(exception))
```
Comment (avanigupta) [inline src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py:372]: Same here, AzureResponseError

**Review by avanigupta — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**